### ISSUE TITLE
Create `EditorControlSocket` Plugin for the Godot Editor

### ISSUE THREAD

**Issue by DaelonSuzuka** [GodotExplorer/godot-tools#318]
I've started working on a plugin to expose editor functionality to a websocket control interface, similar to [obs-websocket](https://github.com/obsproject/obs-websocket). This will allow bidirectional communication and unlock new functionality without trying to shoehorn things into the Language Server. This plugin won't be vscode/godot-tools specific, but would allow other text editors/software to control the godot editor or react to events from it.

Some open issues can be resolved using this interface.

Uses I've thought of so far:
- open a scene file in the editor
- run a specific scene(like pressing F6)
- open a resource
- change/query editor/project settings
- have a scene tree panel in VSCode?
- filesystem stuff
- run "tools" (stuff from the Project > Tools menu)

**Comment by DaelonSuzuka** [comment]
Phase one trials of the `EditorControlSocket` plugin:

https://user-images.githubusercontent.com/18042232/153750270-5761f6c8-5ff9-4528-9cdd-2c2705d2a398.mp4

**Comment by Calinou** [comment]
If this WebSocket server is to be a built-in feature of the Godot editor, please open a proposal on the [Godot proposals](https://github.com/godotengine/godot-proposals) repository :slightly_smiling_face:

**Comment by DaelonSuzuka** [comment]
No, it's not supposed to be a built-in editor feature. The point is that it allows building features for `godot-tools` that do things the Language Server doesn't support yet, or shouldn't support. Additionally, it unlocks `godot-tools` from the editor's release schedule, meaning that bugs can fixed without waiting (potentially years) for patches to make it through to editor releases.

There's already several "serious" use cases in open issues, like #211, which I can fix _today_ using this control socket plugin.
